test/unit/api/java/SolverTest.java

```diff
@@ -102,6 +107,7 @@
 
     Solver slv = new Solver();
     assertThrows(CVC5ApiException.class, () -> slv.mkArraySort(boolSort, boolSort));
+    slv.close();
```

Comment:
Use try-with-resource here.

Reply:
Why are we using a separate solver object here?
It seems that this test mostly checks whether we can use a sort object from solver object a and use it on solver object b... (this is the case for a lot of the tests here). Is this the intention?

Reply:
I think so.  I am just following the corresponding test in `solver_black.cpp`. 
However, I am not a fan of using  `try-with-resource` here. Unlike examples which is written for users, these are unit tests and I don't like the extra indentation that come with `try-with-resources`. 
 I am already using `solver.close` [here](https://github.com/cvc5/cvc5/pull/7436/files#diff-0f2f1183d810770d1917a0457a53e85ddf5cb52c593c50798d8b63ac1ef6cc66R40)
, so I prefer to be consistent here.